src/Shared/SharedUtilities.cs

```diff
@@ -31,6 +32,8 @@ public static class SharedUtilities
 
         private static readonly Lazy<bool> s_isMono = new Lazy<bool>(() => Type.GetType("Mono.Runtime") != null);
 
+        private static readonly Lazy<Process> s_currentProcess = new Lazy<Process>(() => Process.GetCurrentProcess());
```

Comment:
Hmm, wondering if there are could be issues to staticly hanging onto a process reference? Recommendation [here](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process?view=net-6.0#remarks) is to dispose of Process after being used.

Maybe save a string instead something like this?
```suggestion
        private static readonly Lazy<string> s_currentProcessInfo = new Lazy<string>(() => $"{Process.GetCurrentProcess().Id} {Process.GetCurrentProcess().ProcessName}");
```

Reply:
My thought behind was that this is a static variable. So, the dispose would not be necessary. And keeping it there would help in accessing more properties in the future. I am open either way.

Reply:
We want to cache a string, not the Process. Seems less risky with what Peter is proposing.

Reply:
I think we can cache id and name separately and create string one time before the loop. So Id and process can be with no context in the shared utility and used later by anyone. Creating formatted string before the loop should not be costly. Also, the format can be changed anytime.